src/DevBetterWeb.Core/Services/NewMemberService.cs

```diff
@@ -0,0 +1,140 @@
+
+      var message = "Thank you for joining DevBetter! Please click to complete your registration: " + completeRegistrationUrl + "\n";
+
+      await _emailService.SendEmailAsync(inviteEmail, "Welcome to DevBetter!", message);
+    }
+
+    public async Task<string> VerifyValidEmailAndInviteCode(string email, string inviteCode)
+    {
+      var spec = new InvitationByInviteCodeWithEmailSpec(inviteCode);
+
+      string ValidEmailAndInviteCode = "success";
+
+      try
+      {
+        var storedInviteCode = await _repository.GetAsync(spec);
+        if (storedInviteCode == null)
+        {
+          throw new InvitationNotFoundException();
+        }
+        if (!storedInviteCode.Active)
+        {
+          throw new InvitationNotActiveException();
+        }
+        if (storedInviteCode.Email == null)
+        {
+          throw new InvalidEmailException();
+        }
+      }
+      catch(Exception e)
+      {
+        ValidEmailAndInviteCode = "Invalid email or invite code: " + e.GetType().ToString();
+      }
+
+      return ValidEmailAndInviteCode;
+    }
+
+    public async Task<Member> MemberSetup(string userId, string firstName, string lastName, string inviteCode)
+    {
+      Member member = CreateNewMember(userId, firstName, lastName);
+      int memberId = member.Id;
+      await AddUserToMemberRole(userId);
+
+      var spec = new InvitationByInviteCodeWithSubscriptionIdSpec(inviteCode);
+
+      var invite = await _repository.GetAsync(spec);
+      var subscriptionId = invite.PaymentHandlerSubscriptionId;
+
+      var subscriptionStart = _paymentHandlerSubscription.GetStartDate(subscriptionId);
+      var subscriptionEnd = _paymentHandlerSubscription.GetEndDate(subscriptionId);
+
+      CreateSubscriptionForMember(memberId, subscriptionStart, subscriptionEnd);
+
+      // Member has now been created and set up from the invite used. Invite should now be deactivated
+      invite.Deactivate();
+
+      return member;
+    }
+
+
+    private Member CreateNewMember(string userId, string firstName, string lastName)
+    {
+      Member member = new Member(userId);
+      member.UpdateName(firstName, lastName);
+
+      return member;
+    }
+
+    private async Task AddUserToMemberRole(string userId)
+    {
+      var roleName = "Members";
+
+      await _userRoleMembershipService.AddUserToRoleAsyncByRoleName(userId, roleName);
+
+    }
+
+    private Subscription CreateSubscriptionForMember(int memberId, DateTime subscriptionStart, DateTime subscriptionEnd)
+    {
+      var subscription = new Subscription();
+      subscription.MemberId = memberId;
+      DateTimeRange dates = new DateTimeRange(subscriptionStart, subscriptionEnd);
+      subscription.Dates = dates;
+      return subscription;
+    }
+
+    private string GetRegistrationUrl(string inviteCode, string inviteEmail)
+    {
+      var url = $"https://devbetter.com/Identity/Account/NewMemberRegister/{inviteCode}/{inviteEmail}";
```

Comment:
Careful using the full domain here since it will make it harder to test or use in different environments or use in a container, etc. Usually you can have the domain passed in or read from configuration. This is fine for now just FYI.